### ISSUE TITLE
Update lock is generating inconsistent lock with actually resolved dependencies.

### ISSUE THREAD

**Issue by chali** [nebula-plugins/gradle-nebula-integration#22]
I have a bom in a project and one dependency.

```
dependencyLocking {
    lockAllConfigurations()
}

dependencyRecommendations {
    mavenBom module: 'netflix:custom-bom:1.0'
}

dependencies {
    compile('netflix:common')
}
```

Steps to reproduce:
* I generate lock via `./gradlew dependencies --write-locks`
* I update the bom to a new version `netflix:custom-bom:2.0`
* I update lock via `./gradlew dependencies --update-locks 'netflix:custom-bom'`
* I run `./gradlew clean assemble` and it fails with error like this.

```
Could not determine the dependencies of task ':compileJava'.
> Could not resolve all dependencies for configuration ':compileClasspath'.
   > Dependency lock state for configuration 'compileClasspath' is out of date:
       - Did not resolve 'org.springframework.boot:spring-boot-starter-web:2.0.3.RELEASE' which is part of the lock state
```

Names and examples are just illustrative. I will share a real project privately.

**Comment by ljacomet** [comment]
I managed to reproduce this problem. It is related to ordering, pending edges and the current inability for Gradle to revert a no longer pending edge back to pending. See gradle/gradle#6604.

**Comment by ljacomet** [comment]
All issues that were causing this behavior have been fixed, however they are _not yet_ in Gradle nightly `5.0-20180910235855+0000`

**Comment by ljacomet** [comment]
Gradle nightly `5.0-20180912002106+0000` contains the necessary fixes. However, as indicated, adopting this version requires updates to the native BOM support so that the `platform` or `enforcedPlatform` features are used.

**Comment by ljacomet** [comment]
Any progress on verifying this? Is it blocked by another issue? If yes, which one?

**Comment by chali** [comment]
I used today's nightly `5.0-20181004235847+0000` and I don't think this was resolved. Our original project with this problem used to fail with this error when running `./gradlew dependencies`:
```
> Could not resolve all dependencies for configuration ':compile'.
      > Dependency lock state for configuration 'compile' is out of date:
          - Did not resolve 'io.spring.platform:platform-bom:Cairo-SR2' which is part of the lock state
          - Did not resolve 'org.springframework.boot:spring-boot-dependencies:2.0.3.RELEASE' which is part of the lock state
          - Did not resolve 'org.springframework:spring-framework-bom:5.0.7.RELEASE' which is part of the lock state
          - Did not resolve 'org.springframework.cloud:spring-cloud-dependencies:Finchley.RELEASE' which is part of the lock state
          - Did not resolve 'netflix.bom:runtime-platform-recommendations:4.5.60' which is part of the lock state
```
After using latest nightly and running the same command dependencies are printed but modules mentioned above fails to resolve silently.
```
+--- org.springframework.cloud:spring-cloud-dependencies:Finchley.RELEASE FAILED
+--- org.springframework:spring-framework-bom:5.0.7.RELEASE FAILED
+--- io.spring.platform:platform-bom:Cairo-SR2 FAILED
+--- org.springframework.boot:spring-boot-dependencies:2.0.3.RELEASE FAILED
\--- netflix.bom:runtime-platform-recommendations:4.5.60 FAILED
```

Those modules were part of `netflix:custom-bom:1.0` and removed in `2.0` and when I upgrade to 2.0 and update lock I would expect them to be removed from the lock files. An interesting fact is that all those listed which were in bom are also boms.

**Comment by chali** [comment]
The new error is a different flavor of the inconsistent lock. I'm removing `testable` and I will work on a shareable example with the new error.

**Comment by ljacomet** [comment]
I confirm there remains a edge case that this sample illustrates again. Working on a fix. See gradle/gradle#7060

**Comment by ljacomet** [comment]
gradle/gradle#7073 has been merged and this is now testable with nightly from verison `5.0-20181010235832+0000`

**Comment by rpalcolea** [comment]
This is validated, thanks!
